datafusion/sql/src/parser.rs

```diff
@@ -356,9 +355,12 @@ impl<'a> DFParserBuilder<'a> {
         self
     }
 
-    pub fn build(self) -> Result<DFParser<'a>, ParserError> {
+    pub fn build(self) -> Result<DFParser<'a>, DataFusionError> {
         let mut tokenizer = Tokenizer::new(self.dialect, self.sql);
-        let tokens = tokenizer.tokenize_with_location()?;
+        // Convert TokenizerError -> ParserError
+        let tokens = tokenizer
+            .tokenize_with_location()
+            .map_err(|e| ParserError::TokenizerError(e.to_string()))?;
```

Comment:
this is weird that `TokenizerError` is different than `ParserError::TokenizerErrir` and that this is the right way to convert between them. Nothing to change in this PR, maybe something to make nicer . Ideally I would love to see something like:

```rust
            .map_err(ParserError::From)
```